nodebuilder/share/constructors.go

```diff
@@ -29,9 +30,15 @@ func discovery(cfg Config) func(routing.ContentRouting, host.Host) *disc.Discove
 		return disc.NewDiscovery(
 			h,
 			routingdisc.NewRoutingDiscovery(r),
-			cfg.PeersLimit,
-			cfg.DiscoveryInterval,
-			cfg.AdvertiseInterval,
+			availability.WithPeersLimit[availability.DiscoveryParameters](
+				cfg.PeersLimit,
+			),
+			availability.WithDiscoveryInterval[availability.DiscoveryParameters](
+				cfg.DiscoveryInterval,
+			),
+			availability.WithAdvertiseInterval[availability.DiscoveryParameters](
+				cfg.AdvertiseInterval,
+			),
```

Comment:
Those settings are absolutely required by constructor. I would suggest making mandatory settings to be provided by config struct(that is always in there) and  options to set only optional values. I know we done "only options, no config" in some pkgs previously, but we can reconsider this approach. 

WDYT @renaynay @Wondertan @vgonkivs

Reply:
Sounds gut

Reply:
I suggest we do this in a separate PR, logging into an issue for now https://github.com/celestiaorg/celestia-node/issues/1905